smartcontractkit/external-adapters-js
# [SC-51579] Expanded coverage of redacting secrets in logs

## Closes SC-51579

## Description

Some secrets were getting written to logs so the coverage of redacting secrets was expanded to include custom config properties an adapter may use.

## Changes

- Leveraged pino redact configurations to include config property paths that are secret
- Removed previous implementation of cloning config without secrets prior to logging
- Added new URL param match pattern `/client/i` to cover Tradingeconomics ws URLs

## Steps to Test

1. Find adapter with a custom secret config such as privateKey (Harmony, Ethwrite, etc.)
2. Set the appropriate env vars
3. Start the adapter
4. Submit a request
5. Check logs for the config and ensure the field is [REDACTED]
6. Repeat steps for the other redacted paths

## Quality Assurance
